awscli/clidriver.py

```diff
@@ -516,8 +516,14 @@ def __call__(self, args, parsed_globals):
                                                  self._name)
         self._emit(event, parsed_args=parsed_args,
                    parsed_globals=parsed_globals)
-        call_parameters = self._build_call_parameters(parsed_args,
-                                                      self.arg_table)
+        call_parameters = self._build_call_parameters(
+            parsed_args, self.arg_table)
+
+        if not parsed_globals.paginate and \
+                'PaginationConfig' in call_parameters:
+            raise PaginationError(message="--no-paginate provided along with "
+                                          "pagination arguments.")
```

Comment:
I would to prefer to put this logic [here](https://github.com/aws/aws-cli/blob/develop/awscli/customizations/paginate.py#L164) for the following reasons:
1) It is weird to have this pagination logic right in the middle of the clidriver
2) You can have context as to what pagination parameters when they used ``--no-paginate`` and thus have a better error message.